source/numerics/data_out_resample.cc

```diff
@@ -0,0 +1,178 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_tools.h>
+
+#include <deal.II/numerics/data_out_dof_data.templates.h>
+#include <deal.II/numerics/data_out_resample.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <sstream>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+template <int dim, int patch_dim, int spacedim>
+DataOutResample<dim, patch_dim, spacedim>::DataOutResample(
+  const Triangulation<patch_dim, spacedim> &patch_tria,
+  const Mapping<patch_dim, spacedim> &      patch_mapping)
+  : patch_tria(patch_tria)
+  , patch_dof_handler(patch_tria)
+  , patch_mapping(patch_mapping)
+{}
+
+
+
+template <int dim, int patch_dim, int spacedim>
+void
+DataOutResample<dim, patch_dim, spacedim>::update_mapping(
+  const Mapping<dim, spacedim> &mapping,
+  const unsigned int            n_subdivisions)
+{
+  this->mapping = &mapping;
+  this->indices.clear();
+
+  FE_Q<patch_dim, spacedim> fe(std::max<unsigned int>(1, n_subdivisions));
+  patch_dof_handler.distribute_dofs(fe);
+
+  std::vector<std::pair<types::global_dof_index, Point<spacedim>>> points_all;
+
+  QGaussLobatto<patch_dim> quadrature_gl(fe.degree + 1);
```

Comment:
This would be easier to understand if you just used
```
  const Quadrature<patch_dim> quadrature (fe.unit_support_points());
```
which I think is the same.

Reply:
True :+1: